include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
+         * \param data The value containing the data to be set in the specified key
+         *        note - is passed by const-ref
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Note the type of registry value is determined by the template type T of data given
+         *
+         * \remark Examples of usage - note the template type does not need to be explicitly specified
+         *         wil::reg::set_value(key, L"subkey", L"value_name", 0);
+         *
+         *         wil::reg::set_value(key, L"subkey", L"value_name", L"hello");
+         */
+        template <typename T>
+        void set_value(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const T& data)
+        {
+            const reg_view_details::reg_view regview{ key };
+            return regview.set_value(subkey, value_name, data);
+        }
+
+        /**
+         * \brief Writes a value under a specified key, the registry type based off the templated type passed as data
+         * \tparam T The type capturing the data being set
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param value_name A string specifying the name of the registry value to write
+         *        can be nullptr to write to the unnamed default registry value
+         * \param data The value containing the data to be set in the specified key
+         *        note - is passed by const-ref
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Note the type of registry value is determined by the template type T of data given
+         *
+         * \remark Examples of usage - note the template type does not need to be explicitly specified
+         *         wil::reg::set_value(key, L"value_name", 0);
+         *
+         *         wil::reg::set_value(key, L"value_name", L"hello");
+         */
+        template <typename T>
+        void set_value(HKEY key, _In_opt_ PCWSTR value_name, const T& data)
+        {
+            return ::wil::reg::set_value(key, nullptr, value_name, data);
+        }
+
+        /**
+         * \brief Writes a REG_DWORD value from a uint32_t
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subkey A string to append to the HKEY to attempt to write
+         *        can be nullptr if not needed
+         * \param value_name A string specifying the name of the registry value to write
+         *        can be nullptr to write to the unnamed default registry value
+         * \param data The 32-bit value to write to the specified registry value
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Example usage:

```

Comment:
I don't think these examples are particularly illuminating---they're useful for the templated code, but not for these simple calls. We should document overall usage in the wiki, certainly, but I think these examples just add noise to the code.

Reply:
I tried to be consistent and not assume what people know (and prefer to document in code).
i.e., anyone looking at a function to call can see the verification directly in the comment, 'yep, that confirms my understanding'.
If the feedback is it provides negative value to give examples, we should probably have a discussion.